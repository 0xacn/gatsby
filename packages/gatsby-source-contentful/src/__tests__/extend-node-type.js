const { generateImageSource } = require(`../extend-node-type`)

describe(`contentful extend node type`, () => {
  describe(`generateImageSource`, () => {
    it(`default`, () => {
      const resp = generateImageSource(`test.png`, 420, 210, `webp`, null, {})
      expect(resp.src).toContain(`w=420`)
      expect(resp.src).toContain(`h=210`)
      expect(resp.src).toContain(`fm=webp`)
      expect(resp).toMatchSnapshot()
    })
    it(`supports corner radius`, async () => {
      const resp = generateImageSource(`test.png`, 420, 210, `webp`, null, {
        cornerRadius: 10,
      })
      expect(resp.src).toContain(`r=10`)
      expect(resp).toMatchSnapshot()
    })
    it(`transforms corner radius -1 to max`, async () => {
      const resp = generateImageSource(`test.png`, 420, 210, `webp`, null, {
        cornerRadius: -1,
      })
      expect(resp.src).toContain(`r=max`)
      expect(resp).toMatchSnapshot()
    })
    it(`does not include corner by default`, async () => {
      const resp = generateImageSource(`test.png`, 420, 210, `webp`, null, {})
      expect(resp.src).not.toContain(`r=`)
      expect(resp).toMatchSnapshot()
    })
  })
})
