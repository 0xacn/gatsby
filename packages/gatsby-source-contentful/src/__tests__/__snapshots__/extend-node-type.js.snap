// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`contentful extend node type generateImageSource default 1`] = `
Object {
  "format": "webp",
  "height": 210,
  "src": "test.png?w=420&h=210&fm=webp",
  "width": 420,
}
`;

exports[`contentful extend node type generateImageSource does not include corner by default 1`] = `
Object {
  "format": "webp",
  "height": 210,
  "src": "test.png?w=420&h=210&fm=webp",
  "width": 420,
}
`;

exports[`contentful extend node type generateImageSource supports corner radius 1`] = `
Object {
  "format": "webp",
  "height": 210,
  "src": "test.png?w=420&h=210&fm=webp&r=10",
  "width": 420,
}
`;

exports[`contentful extend node type generateImageSource transforms corner radius -1 to max 1`] = `
Object {
  "format": "webp",
  "height": 210,
  "src": "test.png?w=420&h=210&fm=webp&r=max",
  "width": 420,
}
`;
